instrumentation/opensearch/opensearch-rest-1.0/javaagent/src/test/groovy/OpenSearchRestTest.groovy

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import groovy.json.JsonSlurper
+import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+import org.apache.http.HttpHost
+import org.apache.http.auth.AuthScope
+import org.apache.http.auth.UsernamePasswordCredentials
+import org.apache.http.client.CredentialsProvider
+import org.apache.http.conn.ssl.TrustAllStrategy
+import org.apache.http.impl.client.BasicCredentialsProvider
+import org.apache.http.ssl.SSLContextBuilder
+import org.apache.http.util.EntityUtils
+import org.opensearch.client.Request
+import org.opensearch.client.Response
+import org.opensearch.client.ResponseListener
+import org.opensearch.client.RestClient
+import org.opensearch.testcontainers.OpensearchContainer
+import org.testcontainers.utility.DockerImageName
+import spock.lang.Shared
+
+import javax.net.ssl.SSLContext
+import java.util.concurrent.CountDownLatch
+
+import static io.opentelemetry.api.trace.SpanKind.CLIENT
+import static io.opentelemetry.api.trace.SpanKind.INTERNAL
+
+class OpenSearchRestTest extends AgentInstrumentationSpecification {
```

Comment:
if you don't mind converting this to Java, we are (very) slowly migrating all of the groovy tests to Java

here's a relevant example of a Java tests that uses testcontainers: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/kafka/KafkaIntegrationTest.java

Reply:
Done

Reply:
❤️